capstone/src/main/java/com/google/sps/servlets/SignUpServlet.java

```diff
@@ -0,0 +1,72 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.sps.Objects.Badge;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import java.io.IOException;
+import java.io.PrintWriter;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+
+@WebServlet("/signup")
+public class SignUpServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()) {
+      String email = userService.getCurrentUser().getEmail();
+      String userId = userService.getCurrentUser().getUserId();
+      String first = userService.getCurrentUser().getNickname();
+      String last = "";
+
+      if (first.contains(" ")) {
+        String[] split = first.split(" ");
+        first = split[0];
+        last = split[1];
+      }
```

Comment:
I think I'm a little confused as to what is happening here. Does the UserService only store a user's name as one string? So you need to essentially find the first and last names?
Also, if you are getting the name from the UserService, do you users need to be able to change their name or could we just rely on the name from UserService? If the latter is the case, then we probably don't need to split first and last name

Reply:
Yes, the idea was to split up a user's first and last name. UserService doesn't necessarily return a user's actual name - if they don't have it set, it could just be their email address (which is why I did the check for if the name contains a space). So I would like for users to be able to manually change their name within the app. I think I will end up creating an extra page that users go to upon registering for the first time that will handle that. The reason I wanted first/last name separated was because on the feed page for our mock it greets the user with "Welcome back, [firstname]!" which seems friendlier and less formal than using their full name. 

But I'll try to clear up this code or maybe just leave the UserService name as is until the user goes in to manually change it.